panther_lights/test/integration/panther_lights.test.py

```diff
@@ -0,0 +1,143 @@
+from panther_msgs.srv import SetLEDAnimation
+from sensor_msgs.msg import Image
+from std_srvs.srv import SetBool
+
+
+def generate_test_description():
+
+    led_config_file = (
+        PathJoinSubstitution([FindPackageShare("panther_lights"), "config", "led_config.yaml"]),
+    )
+
+    lights_controller_node = Node(
+        package="panther_lights",
+        executable="lights_controller_node",
+        parameters=[{"led_config_file": led_config_file}],
+    )
+
+    lights_driver_node = Node(
+        package="panther_lights",
+        executable="lights_driver_node",
+    )
+
+    # Start test after 1s
+    delay_timer = launch.actions.TimerAction(
+        period=1.0, actions=[launch_testing.actions.ReadyToTest()]
+    )
+
+    actions = [lights_controller_node, lights_driver_node, delay_timer]
+
+    context = {}
+
+    return (
+        LaunchDescription(actions),
+        context,
+    )
+
+
+class TestNodesIntegration(unittest.TestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        rclpy.init()
+
+    @classmethod
+    def tearDownClass(cls):
+        rclpy.shutdown()
+
+    def setUp(self):
+        self.test_node = rclpy.create_node("test_node")
+        self.led_control_requested = None
+
+        self.led_control_enable_srv = self.test_node.create_service(
+            srv_type=SetBool,
+            srv_name="hardware/led_control_enable",
+            callback=self.led_control_enable_cb,
+            qos_profile=rclpy.qos.qos_profile_services_default,
+        )
+
+        self.set_led_animation_client = self.test_node.create_client(
+            srv_type=SetLEDAnimation,
+            srv_name="lights/set_animation",
+            qos_profile=rclpy.qos.qos_profile_services_default,
+        )
+
+    def tearDown(self):
+        self.test_node.destroy_node()
+
+    def led_control_enable_cb(self, request, response):
+        self.led_control_requested = request.data
+        response.success = True
+        response.message = "LED control enabled"
+        return response
+
+    def test_initialization(self, proc_output):
+        rclpy.spin_until_future_complete(self.test_node, rclpy.task.Future(), timeout_sec=2.0)
+
+        self.assertTrue(self.led_control_requested)
+
+        # Controller initialization
+        proc_output.assertWaitFor("[lights_controller]: Loaded default animations.")
+        proc_output.assertWaitFor("[lights_controller]: Initialized successfully.")
+        # Driver initialization
+        proc_output.assertWaitFor("[lights_driver]: Node constructed successfully.")
+        proc_output.assertWaitFor("[lights_driver]: LED control granted.")
+
+    def request_error_animation(self):
+        request = SetLEDAnimation.Request()
+        request.animation = LEDAnimation(id=LEDAnimation.ERROR)
+        request.repeating = False
+
+        self.set_led_animation_client.wait_for_service(timeout_sec=1.0)
+        self.set_led_animation_client.call_async(request)
+
+    def test_msg_publishers(self):
```

Comment:
Is there a simple way to check msg subscribers? Like check if lights_driver_node is subscribed to frame topics?

Reply:
Added new functionality 👍🏼